tests/test_workers/test_pkg_managers/test_ruby.py

```diff
@@ -0,0 +1,189 @@
+from pathlib import Path
+from textwrap import dedent
+
+import pytest
+
+from cachito.errors import ValidationError
+from cachito.workers.pkg_managers.ruby import GemMetadata, parse_gemlock
+
+
+class TestGemlockParsing:
+    @pytest.mark.parametrize(
+        "file_contents, expected_dependencies",
+        (
+            # GEM dependency
+            (
+                dedent(
+                    """\
+                    GEM
+                      remote: https://rubygems.org/
+                      specs:
+                        zeitwerk (2.5.4)
+
+                    PLATFORMS
+                      ruby
+
+                    DEPENDENCIES
+                      zeitwerk
+                    """
+                ),
+                [GemMetadata("zeitwerk", "2.5.4", "GEM", "https://rubygems.org/")],
+            ),
+            # GIT dependency
+            (
+                dedent(
+                    """
+                    GIT
+                      remote: https://github.com/3scale/ci_reporter_shell.git
+                      revision: 30b30d655512891f56463e5f1fa125ea1f2df886
+                      specs:
+                        ci_reporter_shell (0.1.0)
+                          ci_reporter (~> 2.0)
+
+                    GEM
+                      remote: https://rubygems.org/
+                      specs:
+                        builder (3.2.4)
+                        ci_reporter (2.0.0)
+                          builder (>= 2.1.2)
+
+                    DEPENDENCIES
+                      ci_reporter_shell!
```

Comment:
Why do we expect only one dependency (ci_reporter_shell) in this case? I know next to nothing about Gemfile.lock, but it looks to me like there are 3 here (builder, ci_reporter, ci_reporter_shell).